src/ec2/model.ts

```diff
@@ -4,29 +4,93 @@
+import { DefaultIamClient } from '../shared/clients/iamClient'
 
 export class Ec2ConnectionManager {
-    // Will need the ec2Client for probing errors,
     private ssmClient: SsmClient
-    //private ec2Client: DefaultEc2Client
+    private ec2Client: Ec2Client
+    private iamClient: DefaultIamClient
 
     public constructor(readonly regionCode: string) {
-        this.ssmClient = new SsmClient(this.regionCode)
-        //this.ec2Client = new DefaultEc2Client(this.regionCode)
+        this.ssmClient = this.createSsmSdkClient()
+        this.ec2Client = this.createEc2SdkClient()
+        this.iamClient = this.createIamSdkClient()
+    }
+
+    protected createSsmSdkClient(): SsmClient {
+        return new SsmClient(this.regionCode)
     }
 
-    private async handleStartSessionError(err: AWS.AWSError): Promise<void> {
-        const failureMessage =
-            "SSM: Failed to start session with target instance. Common reasons include: \n 1. SSM Agent not installed on instance. \n 2. The required IAM instance profile isn't attached to the instance.  \n 3. Session manager setup is incomplete."
-        await vscode.window.showErrorMessage(failureMessage)
+    protected createEc2SdkClient(): Ec2Client {
+        return new Ec2Client(this.regionCode)
+    }
 
-        throw new ToolkitError('Start Session Failed. ')
+    protected createIamSdkClient(): DefaultIamClient {
+        return new DefaultIamClient(this.regionCode)
+    }
+
+    protected async getAttachedPolicies(instanceId: string): Promise<IAM.attachedPoliciesListType> {
+        try {
+            const IamRole = await this.ec2Client.getAttachedIamRole(instanceId)
+            const iamResponse = await this.iamClient.listAttachedRolePolicies(IamRole!.Arn!)
+            return iamResponse.AttachedPolicies!
+        } catch (err) {
+            return []
+        }
+    }
+
+    public async hasProperPolicies(instanceId: string): Promise<boolean> {
+        const attachedPolicies = (await this.getAttachedPolicies(instanceId)).map(policy => policy.PolicyName!)
+        const requiredPolicies = ['AmazonSSMManagedInstanceCore', 'AmazonSSMManagedEC2InstanceDefaultPolicy']
+
+        return requiredPolicies.every(policy => attachedPolicies.includes(policy))
+    }
+
+    public async handleStartSessionError(err: AWSError, selection: Ec2Selection): Promise<Error> {
+        const isInstanceRunning = await this.ec2Client.isInstanceRunning(selection.instanceId)
+        const generalErrorMessage = `Unable to connect to target instance ${selection.instanceId} on region ${selection.region}. `
+        const hasProperPolicies = await this.hasProperPolicies(selection.instanceId)
+
+        if (!isInstanceRunning) {
+            throw new ToolkitError(
+                generalErrorMessage +
+                    'Ensure the target instance is running and not currently starting, stopping, or stopped.',
+                { code: 'EC2SSMStatusError' }
```

Comment:
"Error" is redundant in the error code name and is not used in other `code:` cases. Confirmed with : 

    git grep "code: "